cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+    for key, value in all_data.items():
+        cl_value = getattr(opinion_cluster, key)
+        if not cl_value:
+            # Value is empty for key, we can add it directly to the object
+            OpinionCluster.objects.filter(id=cluster_id).update(**{key: value})
+        else:
+            if value != cl_value:
+                # We have different values, update dict
+                changed_values_dictionary[key] = (value, cl_value)
+
+    return changed_values_dictionary
+
+
+def merge_long_fields(
+    cluster_id: int,
+    field_name: str,
+    overlapping_data: Optional[Tuple[str, str]],
+) -> None:
+    """Merge two long text fields
+
+    :param cluster_id: Cluster id to update
+    :param field_name: field name to update in opinion cluster
+    :param overlapping_data: data to compare from harvard and courtlistener
+    :return: None
+    """
+    if overlapping_data:
+        if overlapping_data[0] and overlapping_data[1]:
+            harvard_data, cl_data = overlapping_data[0], overlapping_data[1]
+            # Do some text comparison
+            similarity = get_cosine_similarity(harvard_data, cl_data)
+            if similarity < 0.9:
+                # they are not too similar, choose the larger one
+                if len(harvard_data) > len(cl_data):
+                    OpinionCluster.objects.filter(id=cluster_id).update(
+                        **{field_name: harvard_data}
+                    )
+            else:
+                pass
+                # should we log long data not really being similar?
+
+
+def merge_judges(
+    cluster_id: int,
+    overlapping_data: Optional[Tuple[str, str]],
+) -> None:
+    """Merge overlapping judge values
+
+    :param cluster_id: Cluster id to update
+    :param overlapping_data: data to compare from harvard and courtlistener
+    :return: None
+    """
+
+    if overlapping_data:
+        harvard_data, cl_data = overlapping_data
+
+        # We check if any word in the string is uppercase
+        cl_data_upper = (
+            True if [s for s in cl_data.split(",") if s.isupper()] else False
+        )
+
+        # Get last names keeping case and cleaning the string (We could have
+        # the judge names in capital letters)
+        cl_clean = set(find_all_judges(cl_data))
+        # Get last names in lowercase and cleaned
+        harvard_clean = set(find_all_judges(unidecode(harvard_data)))
+        judges = titlecase(", ".join(find_all_judges(unidecode(harvard_data))))
+
+        if (
+            harvard_clean.issuperset(cl_clean) or cl_data_upper
+        ) and harvard_clean != cl_clean:
+            OpinionCluster.objects.filter(id=cluster_id).update(judges=judges)
+        elif not harvard_clean.intersection(set(find_all_judges(cl_data))):
+            raise JudgeException("Judges are completely different.")
+
+
+def merge_cluster_dates(
+    cluster_id: int,
+    field_name: str,
+    overlapping_data: Optional[Tuple[str | None, date]],
```

Comment:
Why are we passing this as a tuple instead of as two values like we do everywhere else?

Reply:
We return a dict that contains the fields that changed with the field name as a key, and the tuple contains the value from json file and the value from cl. For example: {"judges": ("Fischer, French", "French")} that way we only pass the tuple to the correct function based on field name, in this example we pass the tuple to merge_judges() function